File: coffeetime-parent/RubyGun/src/07_string.rb
# Literals
s = 'Hello World'
ds = "Hello World"

puts 'foo' + 'bar'
puts 'foo' << 'bar'

# Iterator
for c in s.chars
  puts c
end

# Multiple Lines
text = %q{
    1, 2, 3, #{s}
    one, two, three
    "x, "y", "z"
}
dtext = %Q{
    1, 2, 3, #{s}
    one, two, three
    "x, "y", "z"
}
symbol_text = <<NAME
    1, 2, 3, #{s}
    one, two, three
    "x, "y", "z"
NAME
puts text
puts dtext
puts symbol_text

puts <<'foo', <<'bar'
"hello"
foo
'world'
bar


# String Interpolation
name = 'Peter'
str = "name=#{name}, #{
if name.length>10;
else
  name.length
end}"
puts str

single_quote = 'name=#{name}, #{if name.length>10; else name.length end}'
puts single_quote

# Methods

## SubString
puts 'Hello World'[6..9]

## Format
puts 'name=%s&length=%d' % [name, name.length]

## Replace
puts 'Hello World'.gsub!(/o/, '-')

## Times
puts 'a' * 3

File: coffeetime-parent/Python3Gun/src/07_string.py
# Literals
s = 'Hello World'
ds = "Hello World"

print('foo' + 'bar')

# Iterator
for c in s:
    print(c)

# Multiple Lines
text = """
    1, 2, 3 #{s}
    one, two, three
    "x, "y", "z"
"""
print(text)

print('c:\now')
print('c:\\now')
print(r'c:\now')

# Methods

# SubString
print('Hello World'[6:9])
print('Hello World'[:3])

# Format
print('{0} say {1}'.format('Marry', 'Hello'))
print('{name} say {msg}'.format(name='Marry', msg='Hello'))
print('{0:.2f}{1}'.format(1024.1024, 'GB'))

print('%s' % 97)
print('%c %c %c' % (65, 66, 67))
print('%10.2f' % 97.123)